src/sinks/util/http.rs

```diff
@@ -607,34 +607,45 @@ pub fn validate_headers(
 
 /// Request type for use in the `Service` implementation of HTTP stream sinks.
 #[derive(Clone)]
-pub struct HttpRequest {
+pub struct HttpRequest<T: Send> {
     payload: Bytes,
     finalizers: EventFinalizers,
     request_metadata: RequestMetadata,
+    additional_metadata: Option<T>,
 }
 
-impl HttpRequest {
+impl<T: Send> HttpRequest<T> {
     /// Creates a new `HttpRequest`.
     pub fn new(
         payload: Bytes,
         finalizers: EventFinalizers,
         request_metadata: RequestMetadata,
+        additional_metadata: Option<T>,
```

Comment:
Does this need to be optional? It would avoid an `.expect` on data we know at build time to be present. I see on sinks that don't need additional metadata, they simply set the type to `()` which is as easy to pass in as `None`.

Reply:
Good point, I'll update this 👍